modules/statistics/AvgRTPStatsReporter.js

```diff
@@ -519,42 +554,92 @@ export default class AvgRTPStatsReporter {
      * @param {go figure} frameRate
      * @param {boolean} isLocal if the average is to be calculated for the local
      * video or <tt>false</tt> if for remote videos.
+     * @param {VideoType} videoType
      * @return {number|NaN} average FPS or <tt>NaN</tt> if there are no samples.
      * @private
      */
-    _calculateAvgVideoFps(frameRate, isLocal) {
+    _calculateAvgVideoFps(frameRate, isLocal, videoType) {
+        let avgRemoteFps = 0;
         let peerCount = 0;
-        let subFrameAvg = 0;
         const myID = this._conference.myUserId();
 
         for (const peerID of Object.keys(frameRate)) {
             if (isLocal ? peerID === myID : peerID !== myID) {
-                const videos = frameRate[peerID];
-                const ssrcs = Object.keys(videos);
+                const participant
+                    = isLocal
+                        ? null : this._conference.getParticipantById(peerID);
+                const videosFps = frameRate[peerID];
+
+                // Do not continue without participant for non local peerID
+                if ((isLocal || participant) && videosFps) {
+                    const peerAvgFPS
+                        = this._calculatePeerAvgVideoFps(
+                            videosFps, participant, videoType);
+
+                    if (!isNaN(peerAvgFPS)) {
+                        avgRemoteFps += peerAvgFPS;
+                        peerCount += 1;
+                    }
+                }
+            }
+        }
 
-                let peerFpsSum = 0;
-                let peerSsrcCount = 0;
+        return avgRemoteFps / peerCount;
```

Comment:
It would be easier to read if "avgRemoteFps" was named something like "peersFpsSum" (the same as elsewhere in this file).